### PR TITLE
perf: use WalkBuilder::add to avoid loop

### PR DESCRIPTION
**Pull request by brenner8023**
## Summary by CodeRabbit

- **Refactor**
  - Improved directory traversal efficiency in the `find_util_rules` function, leading to better performance and control flow during path addition.

### CODE REVIEW

**Review by HerringtonDarkholme — APPROVED** [review]
Looks good

File: crates/cli/src/config.rs [hunk 1]
```diff
@@ -92,28 +92,30 @@ fn find_util_rules(
     return Ok(GlobalRules::default());
   };
   let mut utils = vec![];
-  // TODO: use WalkBuilder::add to avoid loop
+  let mut walker = WalkBuilder::new(&base_dir);
```
Comment (HerringtonDarkholme) [inline crates/cli/src/config.rs:95]: I don't think walker should iterate through base_dir by default

**Comment by HerringtonDarkholme** [comment]
I will merge it first and change the WalkBuilder

**Comment by HerringtonDarkholme** [comment]
image


Compile error

**Comment by brenner8023** [comment]
> image
> Compile error

Fixed it. My mistake.

**Comment by HerringtonDarkholme** [comment]
@brenner8023 can you also run `cargo fmt` for the commit and push again? Otherwise it will break the linting

File: crates/cli/src/config.rs (hunk 1 above)
Comment (HerringtonDarkholme) [reply crates/cli/src/config.rs:95]: Also by default WalkBuilder::new(base_path) will add base_path to the iteration procedure. I think we should exclude it.

**Comment by brenner8023** [comment]
> @brenner8023 can you also run `cargo fmt` for the commit and push again? Otherwise it will break the linting

formatted

File: crates/cli/src/config.rs [hunk 2]
```diff
@@ -92,28 +92,33 @@ fn find_util_rules(
...
-        .expect("file type should be available for non-stdin")
-        .is_file()
-      {
-        continue;
-      }
-      let path = config_file.path();
-      let file = read_to_string(path)?;
-      let new_configs = from_str(&file)?;
-      utils.push(new_configs);
+  let mut walker = WalkBuilder::new(base_dir.join(util_dirs.get(0).unwrap()));
```
Comment (HerringtonDarkholme) [inline crates/cli/src/config.rs:95]: util_dirs can be empty so using unwarp here isn't correct

**Comment by HerringtonDarkholme** [comment]
Thanks!

**Pull request merged**